DuckDuckGo/LocalNotificationsLogic.swift

```diff
@@ -0,0 +1,256 @@
+        store.notificationsEnabled = enabled
+    }
+    
+    func didEnterApplication(currentDate: Date = Date()) {
+        UIApplication.shared.applicationIconBadgeNumber = 0
+        
+        markOverdueNotificationsAsFired(currentDate: currentDate)
+        
+        if let privacyStatus = store.scheduleStatus(for: .privacy),
+            case ScheduleStatus.scheduled = privacyStatus {
+            cancelPrivacyNotification()
+        }
+    }
+    
+    func didSelectNotification(withIdentifier identifier: String) {
+        if let notification = Notification(rawValue: identifier) {
+            switch notification {
+            case .privacy:
+                Pixel.fire(pixel: .notificationD0Opened)
+            case .homeRow:
+                Pixel.fire(pixel: .notificationD1Opened)
+                delegate?.displayHomeHowInstructions(for: self)
+            }
+        }
+    }
+
+    private func markOverdueNotificationsAsFired(currentDate: Date) {
+        for notification in [Notification.privacy, Notification.homeRow] {
+            if let status = store.scheduleStatus(for: notification),
+                case let ScheduleStatus.scheduled(date) = status,
+                date < currentDate {
+                store.didFire(notification: notification)
+                
+                let pixel: PixelName
+                switch notification {
+                case .privacy:
+                    pixel = .notificationD0Fired
+                case .homeRow:
+                    pixel = .notificationD1Fired
+                }
+                
+                LocalNotifications.shared.checkPermissions { (authorization, alertSettings) in
+                    let parameters = [Keys.authorization: String(authorization.rawValue),
+                                      Keys.alert: String(alertSettings.rawValue)]
+                    Pixel.fire(pixel: pixel, withAdditionalParameters: parameters)
+                }
+            }
+        }
+    }
+    
+    private func cancelPrivacyNotification() {
+        LocalNotifications.shared.cancelNotifications(withIdentifiers: [Notification.privacy.identifier])
+        store.didCancel(notification: .privacy)
+    }
+    
+    func willLeaveApplication() {
+        
+        if variantManager.isSupported(feature: .dayZeroNotification),
+            store.notificationsEnabled,
+            store.scheduleStatus(for: .privacy) == nil {
+            schedulePrivacyNotification()
+        }
+        
+        if variantManager.isSupported(feature: .dayOneNotification),
+            store.notificationsEnabled,
+            store.scheduleStatus(for: .homeRow) == nil {
+            scheduleHomeRowNotification()
+        }
+    }
+        
+    private func schedulePrivacyNotification() {
+        let privacyData = PrivacyReportDataSource()
+        
+        let title: String
+        let body: String
+        if privacyData.trackersCount < Constants.privacyNotificationInfoTreshold {
```

Comment:
Should we have a comment about these inline text to let other readers know we intend to move them to `UserText` or delete them depending on the experiment outcome?